pkg/storage/snapshot/snapshot.go

```diff
@@ -388,25 +388,28 @@ func (ctrl *VMSnapshotController) updateVMSnapshotContent(content *snapshotv1.Vi
 	contentCpy.Status.Error = nil
 
 	if len(deletedSnapshots) > 0 {
-		ready = false
+		created, ready = false, false
 		errorMessage = fmt.Sprintf("VolumeSnapshots (%s) missing", strings.Join(deletedSnapshots, ","))
 	} else if len(skippedSnapshots) > 0 {
-		ready = false
+		created, ready = false, false
 		if vmSnapshot == nil || vmSnapshotDeleting(vmSnapshot) {
 			errorMessage = fmt.Sprintf("VolumeSnapshots (%s) skipped because vm snapshot is deleted", strings.Join(skippedSnapshots, ","))
 		} else {
 			errorMessage = fmt.Sprintf("VolumeSnapshots (%s) skipped because in error state", strings.Join(skippedSnapshots, ","))
 		}
 	} else {
 		for _, vss := range volumeSnapshotStatus {
+			if vss.CreationTime == nil {
+				created = false
```

Comment:
if the creation time is nil then it probably cant be ready right? maybe you can add ready=false here and break

Reply:
Personally not crazy about these sorts of micro optimizations.